lib/Doctrine/ORM/Query/Expr/Func.php

```diff
@@ -73,6 +75,10 @@ public function getArguments()
      */
     public function __toString()
     {
+        if (count($this->arguments) === 0) {
+            return $this->name . '(NULL)';
```

Comment:
wouldn't this return all elements where the value is null for nullable columns? This seems wrong.

Reply:
@beberlei No,
```
Select foo from bar where foo IN (NULL)
```
Return always 0 row. I tried it multiple times.

`foo IN (A, B)` mean  `foo = A or foo = B`
`foo IN (NULL)` mean  `foo = NULL`
`foo = NULL` always returns no result, instead of `foo IS NULL`.

And as explained here: https://github.com/doctrine/orm/issues/8033#issuecomment-589950348, this is already the query created when using
```
$query->andWhere("$alias.foo IN (:bar)");
$query->setParameter('bar', []);
```

Reply:
AFAIK `NULL` is handled that way in all commonly known databases like e.g. PostgreSQL, MySQL, SQLite...

https://dbfiddle.uk/?rdbms=sqlite_3.8&fiddle=ac349cc3ca597516e307d690e7f19d90

Reply:
> use x IS NULL, not x = NULL

![https://t.co/bIJIHAJkGa](https://pbs.twimg.com/media/EHHanAQW4AUzg_1?format=jpg&name=900x900) 

&mdash; 🔎Julia Evans🔍 (@b0rk) <a href="https://twitter.com/b0rk/status/1184968545152786433?ref_src=twsrc%5Etfw">October 17, 2019</a>